src/redirects.js

```diff
@@ -0,0 +1,50 @@
+const getDestination = (ctx, AuthUser, redirectDestination) => {
+  if (typeof redirectDestination === 'function') {
+    return redirectDestination({ ctx, AuthUser })
+  }
+
+  return redirectDestination
+}
+
+export const getRedirectToLoginDestination = (
+  authRedirectDestination,
+  ctx,
+  AuthUser
+) => {
+  if (!authRedirectDestination) {
+    throw new Error(
+      'The "authPageURL" config setting must be set when using `REDIRECT_TO_LOGIN`.'
+    )
+  }
+  const destination = getDestination(ctx, AuthUser, authRedirectDestination)
+
+  if (!destination) {
```

Comment:
Let's do some simple validation on this to ensure it's a non-empty string or an object with "destination" set to a non-empty string rather than just ensuring truthiness.